bundles/org.openhab.binding.neohub/src/test/java/org/openhab/binding/neohub/test/NeohubsocketTests.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.test;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.Test;
+import org.openhab.binding.neohub.internal.NeoHubBindingConstants;
+import org.openhab.binding.neohub.internal.NeoHubConfiguration;
+import org.openhab.binding.neohub.internal.NeoHubSocket;
+import org.openhab.binding.neohub.internal.NeoHubWebSocket;
+
+/**
+ * JUnit for testing web and TCP sockets.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class NeohubsocketTests {
+
+    private boolean runOnline = false;
+
+    @SuppressWarnings("null")
+    @Test
+    void testWebSocket() {
+        if (runOnline) {
+            // create the web socket class
+            NeoHubConfiguration config = new NeoHubConfiguration();
+            config.hostName = "192.168.1.xxx";
```

Comment:
What is actually tested by this test with this configuration?

Reply:
If you set the IP address to a valid address, then it tests actual online communication with a hub on that address. The ‘xxx’ is just a placeholder to bypass online connection testing by default.